components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/SequenceApiServiceImpl.java

```diff
@@ -38,9 +44,19 @@ public Response sequenceGet(String apiName, String label, String apiId, MessageC
         JSONObject responseObj = new JSONObject();
         JSONArray sequencesArray = new JSONArray();
         if (gatewayAPIDTO != null) {
-            if (gatewayAPIDTO.getSequenceToBeAdd() != null) {
+            if (gatewayAPIDTO.getSequenceToBeAdd() != null ) {
+                SequenceAdminServiceProxy sequenceAdminServiceProxy =
+                        new SequenceAdminServiceProxy(gatewayAPIDTO.getTenantDomain());
                 for (GatewayContentDTO sequence : gatewayAPIDTO.getSequenceToBeAdd()) {
-                    sequencesArray.put(sequence.getContent());
+                    try {
+                        if (sequenceAdminServiceProxy.isExistingSequence(sequence.getName())) {
+                            sequencesArray.put(sequenceAdminServiceProxy.getSequence(sequence.getName()));
+                        } else {
+                            log.error(sequence.getName() + " was not deployed in the gateway");
```

Comment:
better to have this in the response

Reply:
https://github.com/wso2/carbon-apimgt/pull/8784/commits/2bef8ac27b023cef0271216e1b2942bc739efacc